### ISSUE TITLE
Concurrent using problem

### ISSUE THREAD

**Issue by maralla** [getsentry/sentry-go#46]
Here is the way I use this package:

1. Init sentry in `init` func once
2. To report an event I use the following code:

```go
hub := sentry.CurrentHub().Clone()
hub.WithScope(func(s *sentry.Scope) {
  // ...
  hub.CaptureException(err)
  // ...
})
```
I use the code in multiple goroutines concurrently.

It seems that this package is not goroutine safe as the running program sometimes crashed with message `concurrent map writes`.

Am I take the wrong way of using the package?

**Comment by kamilogorek** [comment]
What SDK version are you using? I fixed some concurrency issues in `0.2.0` and it shouldn't happen anymore - https://github.com/getsentry/sentry-go/commit/50a4841f8a7b546c03f6461c0cbae085ae268394

Goroutines should own their own hub, as explained here - https://docs.sentry.io/platforms/go/goroutines/

**Comment by james-johnston-thumbtack** [comment]
Might it be related to this more specific issue that I just reported? https://github.com/getsentry/sentry-go/issues/55

**Comment by nawa** [comment]
> What SDK version are you using? I fixed some concurrency issues in `0.2.0` and it shouldn't happen anymore - [50a4841](https://github.com/getsentry/sentry-go/commit/50a4841f8a7b546c03f6461c0cbae085ae268394)
> 
> Goroutines should own their own hub, as explained here - https://docs.sentry.io/platforms/go/goroutines/

@kamilogorek 

Ok, and how to `Flush()` all cloned hubs before application exit?

**Comment by nawa** [comment]
`Hub.Flush()` calls `Transport.Flush()` internally

As I see, after `Hub.Clone()` the transport stays unchanged, so `sentry.Flush()` should flush all cloned hubs with the same transport

**Comment by kamilogorek** [comment]
Correct. Buffering is the responsibility of the transport, which is part of the Client. And it on the other hand is copied by reference to every cloned hub.
